src/sentry/ingest/ingest_consumer.py

```diff
@@ -129,6 +129,16 @@ def process_event(message, projects):
 
     # check that we haven't already processed this event (a previous instance of the forwarder
     # died before it could commit the event queue offset)
+    #
+    # XXX(markus): I believe this code is extremely broken:
```

Comment:
I wouldn't say "extremely" here. It does provide certain protection against cases when the consumer is crashing mid-batch, so it doesn't start processing for "good" events over and over.
It was never meant to be a "universal" deduplication solution.